// @prisma/internals/dist/cli/getSchema.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getSchema_exports = {};
__export(getSchema_exports, {
  getPrismaConfigFromPackageJson: () => import_chunk_BHLFQQOC.getPrismaConfigFromPackageJson,
  getRelativeSchemaPath: () => import_chunk_BHLFQQOC.getRelativeSchemaPath,
  getSchema: () => import_chunk_BHLFQQOC.getSchema,
  getSchemaDir: () => import_chunk_BHLFQQOC.getSchemaDir,
  getSchemaPath: () => import_chunk_BHLFQQOC.getSchemaPath,
  getSchemaPathFromPackageJson: () => import_chunk_BHLFQQOC.getSchemaPathFromPackageJson,
  getSchemaPathFromPackageJsonSync: () => import_chunk_BHLFQQOC.getSchemaPathFromPackageJsonSync,
  getSchemaPathInternal: () => import_chunk_BHLFQQOC.getSchemaPathInternal,
  getSchemaPathSync: () => import_chunk_BHLFQQOC.getSchemaPathSync,
  getSchemaPathSyncInternal: () => import_chunk_BHLFQQOC.getSchemaPathSyncInternal
});
module.exports = __toCommonJS(getSchema_exports);
var import_chunk_BHLFQQOC = require("../chunk-BHLFQQOC.js");
var import_chunk_DUMKIIDL = require("../chunk-DUMKIIDL.js");
var import_chunk_FGKIPLS5 = require("../chunk-FGKIPLS5.js");
var import_chunk_IMQ4NNCQ = require("../chunk-IMQ4NNCQ.js");
var import_chunk_IQB47LU2 = require("../chunk-IQB47LU2.js");
var import_chunk_DJGRU4RH = require("../chunk-DJGRU4RH.js");
var import_chunk_OQQ4P4PG = require("../chunk-OQQ4P4PG.js");
