// @prisma/internals/dist/utils/fs-functional.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var fs_functional_exports = {};
__export(fs_functional_exports, {
  createDirIfNotExists: () => import_chunk_YJYPWUHE.createDirIfNotExists,
  getFilesInDir: () => import_chunk_YJYPWUHE.getFilesInDir,
  getNestedFoldersInDir: () => import_chunk_YJYPWUHE.getNestedFoldersInDir,
  removeDir: () => import_chunk_YJYPWUHE.removeDir,
  removeEmptyDirs: () => import_chunk_YJYPWUHE.removeEmptyDirs,
  removeFile: () => import_chunk_YJYPWUHE.removeFile,
  writeFile: () => import_chunk_YJYPWUHE.writeFile
});
module.exports = __toCommonJS(fs_functional_exports);
var import_chunk_YJYPWUHE = require("../chunk-YJYPWUHE.js");
var import_chunk_SAWC6JNZ = require("../chunk-SAWC6JNZ.js");
var import_chunk_J77ZIDXB = require("../chunk-J77ZIDXB.js");
var import_chunk_P56LTJA3 = require("../chunk-P56LTJA3.js");
var import_chunk_MFGQSSGY = require("../chunk-MFGQSSGY.js");
var import_chunk_OQQ4P4PG = require("../chunk-OQQ4P4PG.js");
